mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ExecutorStateManagerImpl.java

```diff
@@ -317,12 +386,35 @@ public GetClusterUsageResponse getClusterUsage(GetClusterUsageRequest req) {
             } else {
                 usageByGroupKey.put(groupKey, kvState);
             }
+
+            if ((value.isAssigned() || value.isRunningTask()) && value.getWorkerId() != null) {
+                if (pendingJobRequests.getIfPresent(value.getWorkerId().getJobId()) != null) {
+                    List<MachineDefinition> workers = jobIdToMachineDef.getOrDefault(value.getWorkerId().getJobId(), new ArrayList<>());
+                    workers.add(value.getRegistration().getMachineDefinition());
+                    jobIdToMachineDef.put(value.getWorkerId().getJobId(), workers);
+                }
+            }
+
+            if (!pendingCountByGroupKey.containsKey(groupKey)) {
+                pendingCountByGroupKey.put(
+                    groupKey,
+                    getPendingCountyByCores(value.getRegistration().getMachineDefinition().getCpuCores()));
+            }
+        });
+
+        // remove jobs from pending set which have all pending workers
+        jobIdToMachineDef.forEach((jobId, workers) -> {
+            final JobRequirements jobStats = pendingJobRequests.getIfPresent(jobId);
+            if (jobStats != null && jobStats.getTotalWorkers() <= workers.size()) {
+                log.info("Removing job {} from pending requests", jobId);
+                pendingJobRequests.invalidate(jobId);
+            }
         });
 
         GetClusterUsageResponseBuilder resBuilder = GetClusterUsageResponse.builder().clusterID(req.getClusterID());
         usageByGroupKey.forEach((key, value) -> resBuilder.usage(UsageByGroupKey.builder()
             .usageGroupKey(key)
-            .idleCount(value.getLeft())
+            .idleCount(value.getLeft() - pendingCountByGroupKey.get(key))
```

Comment:
Currently, we disable the scaler during the upgrade. With this logic job scheduling might see more interruptions and do you think we need to do some handling between upgrade?